core/services/gateway/common/aggregation/workflow_metadata_aggregator.go

```diff
@@ -127,29 +128,53 @@ func (agg *WorkflowMetadataAggregator) Collect(obs *gateway_common.WorkflowMetad
 	_, ok = agg.observations[digest]
 	if !ok {
 		agg.observations[digest] = &NodeObservations{
-			observation: obs,
-			nodes:       make(StringSet),
+			observation:     obs,
+			nodes:           make(StringSet),
+			firstObservedAt: time.Now(),
```

Comment:
Is time.Now() the best way to do it? Why not have a counter variable in the aggregator object that we increment with each item?